src/content/docs/en/reference/experimental-flags/svg.mdx

```diff
@@ -0,0 +1,187 @@
+---
+title: Experimental SVG optimization
+sidebar:
+  label: SVG optimization
+i18nReady: true
+---
+
+import Since from '~/components/Since.astro'
+
+<p>
+
+**Type:** `object`<br />
+**Default:** `undefined`<br />
+<Since v="5.8.0" />
+</p>
+
+This experimental feature enables automatic optimization of SVG assets using [SVGO](https://svgo.dev/) during build time.
+
+When enabled, all imported SVG files will be optimized for smaller file sizes and better performance while maintaining visual quality. This can significantly reduce the size of your SVG assets by removing unnecessary metadata, comments, and redundant code.
+
+To enable this feature, add the experimental flag in your Astro config:
+
+```js title="astro.config.mjs" ins={4-8}
+import { defineConfig } from "astro/config"
+
+export default defineConfig({
+  experimental: {
+    svg: {
+      optimize: true
+    }
+  }
+})
+```
+
+## Configuration
+
+### `optimize`
+
+**Type:** `boolean`<br />
+**Default:** `true`
+
+Whether to enable SVG optimization using SVGO during build time.
+
+When enabled, all imported SVG files will be optimized for smaller file sizes and better performance while maintaining visual quality.
+
+```js title="astro.config.mjs"
+export default defineConfig({
+  experimental: {
+    svg: {
+      optimize: true
+    }
+  }
+})
+```
+
+### `svgoConfig`
+
+**Type:** `Config` (SVGO configuration object)<br />
+**Default:** `{}`
+
+Configuration object passed directly to SVGO for customizing SVG optimization.
+
+See [SVGO documentation](https://svgo.dev/docs/preset-default/) for available options and plugins.
+
+```js title="astro.config.mjs"
+export default defineConfig({
+  experimental: {
+    svg: {
+      optimize: true,
+      svgoConfig: {
+        plugins: [
+          'preset-default',
+          {
+            name: 'removeViewBox',
+            active: false
+          }
+        ]
+      }
+    }
+  }
+})
+```
+
+## Usage
+
+Once enabled, SVG optimization will automatically apply to all SVG files imported in your project:
```

Comment:
Question: does this work if you import SVGs, but then use an `<svg>` element? Or does this only work if you happen to use the SVG component pattern shown below? Is it more about any imported file, or does it further depend on how it's used?

Reply:
This is for components